model/converter/json/from_domain.go

```diff
@@ -97,18 +97,26 @@ func (fd fromDomain) convertReferences(span *model.Span, preserveParentID bool)
 		length++
 	}
 	out := make([]json.Reference, 0, length)
-	if span.ParentSpanID != 0 && !preserveParentID {
-		out = append(out, json.Reference{
-			RefType: json.ChildOf,
-			TraceID: json.TraceID(span.TraceID.String()),
-			SpanID:  json.SpanID(span.ParentSpanID.String()),
-		})
-	}
+	var parentRefAdded bool
 	for _, ref := range span.References {
-		out = append(out, json.Reference{
+		newRef := json.Reference{
 			RefType: fd.convertRefType(ref.RefType),
 			TraceID: json.TraceID(ref.TraceID.String()),
 			SpanID:  json.SpanID(ref.SpanID.String()),
+		}
+		out = append(out, newRef)
+		if newRef.TraceID == json.TraceID(span.TraceID.String()) &&
+			newRef.SpanID == json.SpanID(span.ParentSpanID.String()) {
+			// Check if the parent reference already exists
+			parentRefAdded = true
+		}
+	}
+	if span.ParentSpanID != 0 && !preserveParentID && !parentRefAdded {
+		// TODO this (wrongly) assumes that the reference type is always ChildOf
```

Comment:
it assumes that correctly, because if ParentSpanID != 0 but there are no other references then that ParentSpanID does refer to child-of type (you may want to leave that as a comment)